CryptoTechReminderSystem.Test/GetLateDevelopersTests.cs

```diff
@@ -0,0 +1,249 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using CryptoTechReminderSystem.DomainObject;
+using CryptoTechReminderSystem.Gateway;
+using CryptoTechReminderSystem.UseCase;
+using FluentAssertions;
+using NUnit.Framework;
+
+namespace CryptoTechReminderSystem.Test
+{
+    public class GetLateDevelopersTests
+    {
+        public class HarvestGatewayStub : ITimesheetAndDeveloperRetriever
+        {
+            public HarvestDeveloper[] Developers { get; set; }
+            
+            public TimeSheet[] TimeSheets { get; set; }
+            
+            public IList<HarvestDeveloper> RetrieveDevelopers()
+            {
+                IList<HarvestDeveloper> result = Developers;
+                return result;
+            }
+
+            public IEnumerable<TimeSheet> RetrieveTimeSheets()
+            {
+                IList<TimeSheet> result = TimeSheets;
+                return result;
+            }
+        }
+
+        private class HarvestGatewaySpy : ITimesheetAndDeveloperRetriever
+        {
+            public bool IsRetrieveDevelopersCalled;
+            public bool IsRetrieveTimeSheetsCalled;
+            public IList<HarvestDeveloper> RetrieveDevelopers()
+            {
+                IsRetrieveDevelopersCalled = true;
+                IList<HarvestDeveloper> result = new List<HarvestDeveloper>();
+              
+                return result;
+            }
+
+            public IEnumerable<TimeSheet> RetrieveTimeSheets()
+            {
+                IsRetrieveTimeSheetsCalled = true;
+                return new List<TimeSheet>();
+            }
+        }
+        
+        public class SlackGatewayStub : IMessageSenderAndRetriever
```

Comment:
Class SlackGatewayStub can also be made private.

Reply:
Resolved in https://github.com/chubberlisk/crypto_tech/pull/9/commits/63a0f618d292276577c14db1bce0659100f6b45c